tests/integration/devfile/cmd_devfile_config_test.go

```diff
@@ -0,0 +1,108 @@
+package devfile
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo devfile config command tests", func() {
+	const (
+		testName      = "testName"
+		testMemory    = "500Mi"
+		testDebugPort = "8888"
+		fakeParameter = "fakeParameter"
+	)
+
+	var project, context, currentWorkingDirectory, originalKubeconfig string
+
+	// Using program command according to cliRunner in devfile
+	cliRunner := helper.GetCliRunner()
+
+	// This is run after every Spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		context = helper.CreateNewContext()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		originalKubeconfig = os.Getenv("KUBECONFIG")
+		helper.LocalKubeconfigSet(context)
+		project = cliRunner.CreateRandNamespaceProject()
+		currentWorkingDirectory = helper.Getwd()
+		helper.Chdir(context)
+	})
+
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		cliRunner.DeleteNamespaceProject(project)
+		helper.Chdir(currentWorkingDirectory)
+		err := os.Setenv("KUBECONFIG", originalKubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		helper.DeleteDir(context)
+	})
+
+	Context("When executing config view", func() {
+		It("Should view all default parameters", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs")
+			output := helper.CmdShouldPass("odo", "config", "view")
+			wantOutput := []string{
+				"nodejs",
+				"Ports",
+				"Memory",
+			}
+			helper.MatchAllInOutput(output, wantOutput)
+		})
+	})
+
+	Context("When executing config set", func() {
```

Comment:
Instead of two Context, I would have added two more extra steps with this Context itself saying "When executing config set and unset" 
This will reduce the duplicate steps and you get both the content in one.

Reply:
Done